### PR TITLE
Version 2.0

### PR DESCRIPTION
**Pull request by ChristianRiesen**
Trying to get this to a better standard, requiring php 7.4

### CODE REVIEW

**Comment by reedy** [comment]
I note for "us" (MediaWiki/Wikimedia), 7.2 would be appreciated

File: tests/bootstrap.php [hunk 1]
```diff
@@ -1,5 +1,5 @@
 <?php
```
Comment (reedy) [inline tests/bootstrap.php:1]: Is this file actually needed for anything? It... shouldn't be...

File: tests/Base32Test.php [hunk 2]
```diff
@@ -1,51 +1,60 @@
...
 
-namespace Base32;
+declare(strict_types=1);
+
+namespace Base32\Tests;
 
 use Base32\Base32;
+use PHPUnit\Framework\TestCase;
+
+class Base32Test extends TestCase
```
Comment (reedy) [inline tests/Base32Test.php:10]: Probably worth adding some @covers

File: composer.json [hunk 3]
```diff
@@ -14,12 +14,12 @@
         }
     ],
     "require": {
-        "php": ">=5.3"
+        "php": "^7.4"
```
Comment (reedy) [inline composer.json:17]: Most tend to do >=version atm...

So `>=7.4` would be fine (though, I would prefer `>=7.2`)

Or `^7.4||^8.0` would be better (or well, `^7.2||^8.0`)

**Review by reedy — CHANGES_REQUESTED** [review]

File: composer.json [hunk 4]
```diff
@@ -14,12 +14,12 @@
...
     },
     "require-dev": {
-        "php": ">=5.6",
-        "phpunit/phpunit": "^5.0",
-        "satooshi/php-coveralls": "^1.0"
+        "friendsofphp/php-cs-fixer": "^2.17",
+        "phpstan/phpstan": "^0.12",
+        "phpunit/phpunit": "^9.5"
```
Comment (reedy) [inline composer.json:22]: `^8.5.13||^9.5.0` should work fine to keep 7.2 support, without additiona changes...

**Comment by ChristianRiesen** [comment]
@reedy Thanks for your input. I wasn't aware MediaWiki is using this :)
I incorporated all your inputs and got the builds to work with all the versions (7.2 tests with old phpunit, everything else with the newer one). There was no functional need to make it require 7.4 and so I made sure it keeps working and being tested on 7.2 as well. Also getting tested on 8.0 so a move to that should be fine as well.

If you have more inputs, please send them my way, I'll leave this PR open for another day or two before merging and tagging. Not even a BC break happening here, so not a v2 after all :)

**Comment by reedy** [comment]
> Not even a BC break happening here, so not a v2 after all :)

Well... YMMV on whether bumping PHP versions is a semver major change. Different groups argue different ways.

I would say as you're dropping 5.x, it's probably easier/better to just bump the major version; doing so is cheap. Rather than people having something like `^1.3`, still being on PHP 5.3 and wondering why 1.4 won't immediately work (though, composer should take of care of that, I think) :)

I'll have another look/review first

**Review by reedy — CHANGES_REQUESTED** [review]
My only irk is the yoda conditionals - `'' == $foo` rather than `$foo == ''`, and more so the changing to them.. I understand this is personal style, and some projects (at least [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/php/#yoda-conditions)) do actively suggest doing it.

Blockers: Couple of wording changes needed. And then consistency on whether `\` is used before global functions. Otherwise LGTM I think :)

File: src/Base32.php [hunk 5]
```diff
@@ -136,11 +122,30 @@ public static function decode($base32String)
...
+     * @param string $binaryString String to chunk
+     * @param int    $bits         Number of bits per chunk
+     *
+     * @return array<string>
+     */
+    private static function chunk(string $binaryString, int $bits): array
+    {
+        $binaryString = chunk_split($binaryString, $bits, ' ');
+
+        if (' ' === substr($binaryString, \strlen($binaryString) - 1)) {
```
[the remark this answers is not in the document]
Comment (ChristianRiesen) [reply src/Base32.php]: It adds some speed, so why not use it consistently for everything. My IDE only flags certain functions where it is proven to add more speed, but might as well add it everywhere.

File: README.md [hunk 6]
```diff
@@ -45,9 +37,9 @@ Have a RFC compliant Base32 encoder and decoder. The implementation could be imp
 Requirements
 ------------
 
-PHP 5.3 to 5.6 or 7.0+
+Works on PHP 7.2 and later, including PHP 8.0.
 
-If you want to run the tests, PHPUnit 5.0+ or up is required. Tests require PHP 5.6 or 7.0+.
+Tests run on PHPUnit 9.5, with PHP 7.3 and later. For PHP 7.2 uses an older PHPUnit version.
```
[the remark this answers is not in the document]
Comment (ChristianRiesen) [reply README.md]: Changed

**Comment by ChristianRiesen** [comment]
Thanks again for the inputs @reedy 
I agree, consistency makes more sense, so since it can add performance, might as well do it. Also added a check to cs fixer for that for the future.
Also changed the README. Should be slightly better.
Regarding semver, I was of the opinion to make a major version change when changing minimum php version. After a discussion with a co worker I changed my mind. Composer takes care of that, as that is a dependency. The code itself (or it's interfaces really) should dictate a major change or not in the version. Thats at least my thinking here.
On yoda, I was not a fan at first, though saw the logical reasoining behind it. I have since starting to use it more seriously seen multiple bugs that could have been prevented by using it, so I'm convinced. It makes the conditions slightly harder to read, but this also points out to me when a condition is too large to properly understand quickly. Should have no influence of the functionality of the library though :)

I'll merge and tag then. More inputs always welcome. Now I need to fix my otp library :)

**Pull request merged**